### ISSUE TITLE
Reference repos files in a repos file

### ISSUE THREAD

**Issue by ghost** [dirk-thomas/vcstool#241]
Consider the following situation:

- There 15 repositories a team is working on
- 10 of them are 'core', as in everyone uses them
- 5 of them are subproject related
- There are repos files for each, named project1, project2, project3, project4, project5
- Each repos files includes all 10 core repos

It would be nice to only define the core config once and include or inline it somehow in each project specific definition instead of repeating the same 50 lines 5 times.

Has this been considered before?

**Comment by christophebedard** [comment]
I worked on this exact feature and have a working prototype, see here: https://github.com/dirk-thomas/vcstool/pull/148

Unfortunately, it didn't go anywhere as far as merging it into this repo.

**Comment by ghost** [comment]
> I worked on this exact feature and have a working prototype, see here: #148
> 
> Unfortunately, it didn't go anywhere as far as merging it into this repo.

Thanks, I'll take a look

**Comment by christophebedard** [comment]
It might be possible to extract this feature into a separate tool that takes the "extension" repos file as an input and outputs the combined repos file. Then you could just pipe the output into `vcs`.

**Comment by esteve** [comment]
@christophebedard was there anything left to do in #148 ? Or is it a matter of getting more feedback from @dirk-thomas ? Looks to me that your PR already achieves the goal you wanted, right?

**Comment by christophebedard** [comment]
@esteve yes the PR is pretty complete. I think the repo is in light maintenance mode, since Dirk has stopped needing this tool. Maybe @dirk-thomas would be open to letting other people take over maintainership?

**Comment by esteve** [comment]
@christophebedard thanks for the info. Given that `vcstool` is such a fundamental tool for the entire ROS ecosystem, perhaps it'd make sense to move the repository to the @ros organization, what do you think @dirk-thomas ?

**Comment by christophebedard** [comment]
@esteve should probably bring this up in a https://github.com/ros2/ros2 issue. I think Dirk is enjoying life these days :smile:

**Comment by esteve** [comment]
@christophebedard I've created https://github.com/dirk-thomas/vcstool/issues/242, but if we don't get a response, I'll move the discussion to Discourse

**Comment by katettudelft** [comment]
Was there ever any progress on this issue? I've got a very similar use case to the one above.

**Comment by christophebedard** [comment]
No progress unfortunately.

**Comment by esteve** [comment]
@christophebedard I opened #242 a long time ago, but didn't get a response from @dirk-thomas . Given that you showed interest in helping maintain `vcstool`, it'd be good to move or fork the repository somewhere else and fom a group.

cc @nuclearsandwich
